### PR TITLE
[Fix] Fix the bug that vit cannot load pretrain properly when using i…

### PR DESCRIPTION
**Pull request by RockeyCoss**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

 Fix the bug that vit cannot load pretrain properly when using init_cfg to specify the pretrain scheme

## Modification

 Fix the bug that vit cannot load pretrain properly when using init_cfg to specify the pretrain scheme

### CODE REVIEW

File: tests/test_models/test_backbones/test_vit.py [hunk 1]
```diff
@@ -118,3 +118,44 @@ def test_vit_backbone():
     feat = model(imgs)
     assert feat[0][0].shape == (1, 768, 14, 14)
     assert feat[0][1].shape == (1, 768)
+
+
+def test_vit_init():
+    path = 'PATH_THAT_DO_NOT_EXIST'
+    # Test all combinations of pretrained and init_cfg
```
Comment (Junjun2016) [inline tests/test_models/test_backbones/test_vit.py:125]:
```suggestion
    # Test all combinations of pretrained and init_cfg
    # pretrained=None, init_cfg=None
```

File: tests/test_models/test_backbones/test_vit.py [hunk 2]
```diff
@@ -118,3 +118,44 @@ def test_vit_backbone():
...
+    model.init_weights()
+
+    model = VisionTransformer(
+        pretrained=None, init_cfg=dict(type='Pretrained', checkpoint=path))
+    assert model.init_cfg == dict(type='Pretrained', checkpoint=path)
+    # Test loading a checkpoint from an non-existent file
+    with pytest.raises(OSError):
+        model.init_weights()
+
+    model = VisionTransformer(pretrained=None, init_cfg=123)
```
Comment (Junjun2016) [inline tests/test_models/test_backbones/test_vit.py:137]: Add comment for this case.

File: tests/test_models/test_backbones/test_vit.py [hunk 3]
```diff
@@ -118,3 +118,44 @@ def test_vit_backbone():
...
+
+    with pytest.raises(TypeError):
+        model = VisionTransformer(pretrained=123, init_cfg=None)
+
+    with pytest.raises(AssertionError):
+        model = VisionTransformer(
+            pretrained=123, init_cfg=dict(type='Pretrained', checkpoint=path))
+
+    with pytest.raises(AssertionError):
+        model = VisionTransformer(pretrained=123, init_cfg=123)
```
Comment (Junjun2016) [inline tests/test_models/test_backbones/test_vit.py:161]: Add comment for each case.

**Review by Junjun2016 — APPROVED** [review]
Please resolve some comments.

**Review by xvjiarui — APPROVED** [review]
LGTM

**Pull request merged**